### PR TITLE
commit

### PR DESCRIPTION
**Pull request by Anchalexcellence**

### CODE REVIEW

File: app/common/getDataFactory.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 angular.module('batApp').factory('getDataFactory', (configuration, $resource) => ({
   getOutletViewData () {
-    return $resource(configuration.apihost + '/outlet/get/-1/1')
+    return $resource(configuration.apihost + '/outlet/get/1/13')
```
Comment (mageexxcel) [inline app/common/getDataFactory.js]: why is this 1/13 hard coded

File: app/common/getDataFactory.js [hunk 2]
```diff
@@ -12,5 +12,18 @@ angular.module('batApp').factory('getDataFactory', (configuration, $resource) =>
...
-  }
+  },
+  getTmeViewData () {
+    return $resource(configuration.apihost + '/tme/get/-1/1')
+  },
+  addTme () {
+    return $resource(configuration.apihost + '/tme/register')
+  },
+  getSkuViewData () {
+    return $resource(configuration.apihost + '/get/sku/1/10')
```
Comment (mageexxcel) [inline app/common/getDataFactory.js]: what is 10

File: app/components/addTme/addTmeController.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      vm.err = false
+
+      vm.save = function (data) {
+        getDataFactory.addTme().save(data).$promise
+                .then((response) => {
+                  if (!response.error) {
+                    $state.go('menuTemplate.tmeViewTable')
+                  } else {
+                    vm.err = true
+                    vm.errMsg = response.error.message
```
Comment (mageexxcel) [inline app/components/addTme/addTmeController.js]: we need to have global error messages. not in every api.   

in interceptor need to capture error message and show in all api error messsages

File: app/components/outlets/viewOutletController.js [hunk 4]
```diff
@@ -21,6 +21,7 @@ angular.module('batApp').controller('viewOutletController', function (getDataFac
         .then((response) => {
           if (response.error) {} else {
             angular.forEach(response, (value, key) => {
+              value.data.updatedAt = moment(value.data.updatedAt).format('DD MMMM, YYYY')
```
Comment (mageexxcel) [inline app/components/outlets/viewOutletController.js]: do all this in service. controller should be small

File: app/components/sku/viewSkuController.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+  }
+  getDataFactory.getSkuViewData().query().$promise
+     .then((response) => {
+       if (response.error) {} else {
+         angular.forEach(response, (value, key) => {
+           value.sku.createdAt = moment(value.sku.createdAt).format('DD MMMM, YYYY')
+
+           vm.skuList.push([value.sku.id, value.sku.productname, value.sku.brand_id, value.sku.createdAt, value.sku.bat_id, value.sku.basepoint])
```
Comment (mageexxcel) [inline app/components/sku/viewSkuController.js]: do all this in service. controller should get final data to be used

File: app/common/getDataFactory.js (hunk 1 above)
Comment (Anchalexcellence) [reply app/common/getDataFactory.js]: getting data with api '/outlet/get/-1/1'

File: app/common/getDataFactory.js (hunk 2 above)
Comment (Anchalexcellence) [reply app/common/getDataFactory.js]: replace 10 by 100 as api to get sku data is '/get/sku/1/100'

File: app/components/outlets/viewOutletController.js (hunk 4 above)
Comment (Anchalexcellence) [reply app/components/outlets/viewOutletController.js]: make service and send final data to controller

File: app/components/sku/viewSkuController.js (hunk 5 above)
Comment (Anchalexcellence) [reply app/components/sku/viewSkuController.js]: make service and send final data to controller

File: app/components/addTme/addTmeController.js (hunk 3 above)
Comment (Anchalexcellence) [reply app/components/addTme/addTmeController.js]: sir these are form validation error i have removed from here and used angular validation

File: app/batApp.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-angular.module('batApp', ['ui.router', 'ngResource', 'ngStorage', 'ngMaterial'])
+angular.module('batApp', ['ui.router', 'ngResource', 'ngStorage', 'ngMaterial', 'angularMoment'])
```
Comment (mageexxcel) [inline app/batApp.js:3]: where is this used? 
angularMoment

File: app/common/tables.html [hunk 7]
```diff
@@ -2,30 +2,32 @@
...
+               <th  ng-if="data.Type=='SUBMIT'" ng-repeat="data in tabledetails.button">
+                  <button type="submit" class="btn btn-primary"> {{data.Type}}
+               </th>
+            </tr>
+         </thead>
+         <tbody>
+            <tr ng-repeat="viewdata in tablebody">
+               <td ng-repeat="value in viewdata track by $index">{{value}}</td>
+               <td></td>
+               <td><a ng-class="tabledetails.alink.Text=='Edit' ?'linkWithPadding' :xyz">{{tabledetails.alink.Text}}</a>
```
Comment (mageexxcel) [inline app/common/tables.html]: what is xyz?

File: app/batApp.js (hunk 6 above)
Comment (Anchalexcellence) [reply app/batApp.js:3]: sir in services  to format the date to show on the view

**Pull request merged**